e2e/libs/node_exec/node_exec.py

```diff
@@ -80,11 +81,20 @@ def issue_cmd(self, node_name, cmd):
             stdout=True,
             tty=False
         )
+        logging(f"Issued command: {cmd} on {node_name} with result {res}")
         return res
 
     def launch_pod(self, node_name):
         if node_name in self.node_exec_pod:
-            return self.node_exec_pod[node_name]
+            for i in range(DEFAULT_POD_TIMEOUT):
+                pod = self.core_api.read_namespaced_pod(
+                        name=node_name,
+                        namespace=self.namespace
+                      )
+                if pod is not None and pod.status.phase == 'Running':
+                    break
+                time.sleep(DEFAULT_POD_INTERVAL)
+            return pod
```

Comment:
Perhaps we should consider verifying this elsewhere. Is there a specific reason for having this here? Shouldn't we be cleaning up after tests?

Reply:
After node reboot or other k8s related negative factors, the pod could not be in `Running` state even though it's created before. Need to wait for it recovering.

Reply:
The function's name suggests that it should create a pod. If we can identify the relevant factors, the subsequent step should involve waiting for the pod to recover. This approach enhances transparency in the test steps, reducing ambiguity.

Reply:
Contributors don't even have to know there is such an execution pod running.

Reply:
It's better for the tests to have transparent steps.

Reply:
Actually, I think I've misunderstood you. What I meant to ask is why there would be an existing pod running during `launch_pod`.

Reply:
> why there would be an existing pod running during `launch_pod`

Like a singleton class, when you instantiate it, internally it checks whether there's an instance exists, if so, just returns this existing one, otherwise creates a new one for you.

> why we are reusing the pod

A node execution pod, as a delegate, is meant to be reused throughout the whole test to provide a simpler way to execute commands on a node without ssh hostname/key configuration.